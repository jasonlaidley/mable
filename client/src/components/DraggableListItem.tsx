import * as React from 'react';
import { Draggable } from 'react-beautiful-dnd';
import ListItemText from '@mui/material/ListItemText';
import Typography from '@mui/material/Typography';
import ListItemButton from '@mui/material/ListItemButton';
import { purple } from '@mui/material/colors';

// Define the DraggableListItemProps type
type DraggableListItemProps = {
  person: {
    id: string;
    name: string;
    age: number;
    location: string;
  };
  index: number;
};

const DraggableListItem = ({person, index}: DraggableListItemProps) => {
  return (
    <Draggable key={person.id} draggableId={person.id} index={index}>
      {(provided) => (
        <ListItemButton ref={provided.innerRef} {...provided.draggableProps} {...provided.dragHandleProps} divider={true} sx={{bgcolor: purple[50]}}>
          <ListItemText 
            primary={person.name} 
            secondary={
              <React.Fragment>
                <Typography sx={{display: 'inline'}} component="span" variant="body2" color="text.primary">
                  {person.age}
                </Typography>
                {` - ${person.location}`}
              </React.Fragment>
            } 
          />
        </ListItemButton>
      )}
    </Draggable>
  );
};

export default DraggableListItem;
