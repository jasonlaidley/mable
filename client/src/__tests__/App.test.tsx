import * as React from 'react';
import renderer from 'react-test-renderer';
import { DragDropContext, Droppable } from 'react-beautiful-dnd';
import DraggableListItem from '../components/DraggableListItem';

const person = {
  id: '1',
  name: 'Bruce Wayne',
  age: 40,
  location: 'Brisbane'
};

const index = 0;

it('renders correctly', () => {
  const tree = renderer.create(
    <DragDropContext onDragEnd={() => {}}>
      <Droppable droppableId="droppable">
        {(provided) => (
          <div {...provided.droppableProps} ref={provided.innerRef}>
            <DraggableListItem person={person} index={index} key={person.id} />
            {provided.placeholder}
          </div>
        )}
      </Droppable>
    </DragDropContext>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});