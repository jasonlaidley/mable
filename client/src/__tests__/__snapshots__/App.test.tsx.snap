// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-rbd-droppable-context-id="0"
  data-rbd-droppable-id="droppable"
>
  <div
    aria-describedby="rbd-hidden-text-0-hidden-text-0"
    className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-divider MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-divider css-15xraz-MuiButtonBase-root-MuiListItemButton-root"
    data-rbd-drag-handle-context-id="0"
    data-rbd-drag-handle-draggable-id="1"
    data-rbd-draggable-context-id="0"
    data-rbd-draggable-id="1"
    draggable={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    onTransitionEnd={null}
    role="button"
    style={
      {
        "transform": null,
        "transition": null,
      }
    }
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Bruce Wayne
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body2 css-1phucj-MuiTypography-root"
        >
          40
        </span>
         - Brisbane
      </p>
    </div>
  </div>
</div>
`;
